Sources/Plasma/FeatureLib/pfPython/plPythonCallable.h

```diff
@@ -62,16 +62,93 @@ plProfile_Extern(PythonUpdate);
 
 namespace plPython
 {
+    class KwArg
+    {
+        pyObjectRef fName;
+        pyObjectRef fValue;
+
+    public:
+        KwArg() = delete;
+        KwArg(const KwArg&) = delete;
+
+        template<size_t _Sz, typename _ValueT>
+        KwArg(const char (&name)[_Sz], _ValueT&& value)
+            : fName(PyUnicode_FromStringAndSize(name, _Sz - 1)),
+              fValue(ConvertFrom(std::forward<_ValueT>(value)))
+        { }
+
+        KwArg(KwArg&& move) noexcept
+            : fName(std::move(move.fName)),
+              fValue(std::move(move.fValue))
+        {
+        }
+
+        PyObject* ReleaseName() { return fName.Release(); }
+        PyObject* ReleaseValue() { return fValue.Release(); }
+    };
+
+    inline PyObject* ConvertFrom(KwArg&& kwarg)
+    {
+        return kwarg.ReleaseValue();
+    }
+
     namespace _detail
     {
+        template <typename... Args>
+        struct PositionalAfterKwArg;
+
+        template <>
+        struct PositionalAfterKwArg<> : std::false_type
+        {
+        };
+
+        template <typename First, typename... Rest>
+        struct PositionalAfterKwArg<First, Rest...>
+        {
+            static constexpr bool value = [] {
+                using T = std::decay_t<First>;
+                if constexpr (std::is_same_v<T, KwArg>) {
+                    return (... || !std::is_same_v<std::decay_t<Rest>, KwArg>);
+                } else {
+                    return PositionalAfterKwArg<Rest...>::value;
+                }
+            }();
+        };
+
+        template <typename Arg>
+        inline void InsertKwName(PyObject* kwlist, Py_ssize_t& i, Arg& arg)
+        {
+            using T = std::decay_t<decltype(arg)>;
+            if constexpr (std::is_same_v<T, KwArg>)
+                PyTuple_SET_ITEM(kwlist, i++, arg.ReleaseName());
```

Comment:
We're in a helper function here, so the increment doesn't need to be inline. (Style preference I suppose, but I'm not a fan of increments/decrements "hidden" inside a larger expression, unless necessary.)

```suggestion
            if constexpr (std::is_same_v<T, KwArg>) {
                PyTuple_SET_ITEM(kwlist, i, arg.ReleaseName());
                i++;
            }
```